oioioi/contests/utils.py

```diff
@@ -637,3 +635,47 @@ def has_view_permission(self, request, obj=None):
             return True
 
     return ArchivedInlineWrapper
+
+
+def get_problem_statements(request, controller, problem_instances):
```

Comment:
it would be a nightmare to maintain this, what about returning list of dicts?

Reply:
Can you elaborate please?

Reply:
In this function you return tuple with very specific order of tuple-elements. This ordering is repeated in two distinct places and the code would fail if they will be out-of-sync.

I suggest to have a get_problem_statements(..) that returns a list of dictionaries i.e.
```
[
  {"problem_instance": pi1, "statement_visible": True, "round_end_time": datetime(...), ...},
  {"problem_instance": pi2, "statement_visible": False, "round_end_time": datetime(...), ...},
   ...
]
```

With this change you still need to maintain sync between different fragments of code, but at least you will get an error in case of mispelled key.

Reply:
Other then this change I'm happy to merge the PR.